File: api/auth-me.js
import { getAuthFromRequest, sendJson } from './auth-shared.js';

export default async function handler(req, res) {
    if (req.method !== 'GET') return sendJson(res, 405, { error: 'Method not allowed' });

    const auth = getAuthFromRequest(req);
    if (!auth || !auth.userId) return sendJson(res, 401, { error: 'Not authenticated' });

    return sendJson(res, 200, { user: { id: auth.userId, username: auth.username } });
}

File: api/auth_me.py
from flask import Flask, request
from api.auth_utils import get_auth_from_request, send_json

app = Flask(__name__)


@app.route('/api/auth/me', methods=['GET'])
def me():
    auth = get_auth_from_request(request)
    if not auth or not auth.get('userId'):
        return send_json({'error': 'Not authenticated'}, 401)
    return send_json({
        'user': {
            'id': auth['userId'],
            'username': auth.get('username')
        }
    })
